k2/csrc/ragged_ops_inl.h

```diff
@@ -667,6 +668,32 @@ Ragged<T> CreateRagged2(const std::vector<std::vector<T>> &vecs) {
   return Ragged<T>(shape, values_array);
 }
 
+template <typename T>
+Array2<T> PadRagged(Ragged<T> &src, T padding_value) {
+  NVTX_RANGE(K2_FUNC);
+  K2_CHECK_EQ(src.NumAxes(), 2);
+  ContextPtr &c = src.Context();
+  int32_t row_num = src.Dim0(),
+          col_num = src.shape.MaxSize(1);
+  Array2<T> res(c, row_num, col_num);
+  int32_t elem_stride0 = res.ElemStride0();
```

Comment:
Array2Accessor is for this particular case. Please refer to
https://github.com/k2-fsa/k2/blob/069425e301472e7ea31ea982ba2a943ac5fcb649/k2/csrc/intersect_dense_pruned.cu#L658
and
https://github.com/k2-fsa/k2/blob/069425e301472e7ea31ea982ba2a943ac5fcb649/k2/csrc/intersect_dense_pruned.cu#L681

Its `operator()(int, int)` returns an lvalue reference.